### ISSUE TITLE
Build error when adding Fragment: Conflict with dependency 'com.android.support:support-annotations'

### ISSUE THREAD

**Issue by misaochan** [commons-app/apps-android-commons#709]
When I create a new Fragment subclass, the Gradle sync fails and I get this error:

> Error:Conflict with dependency 'com.android.support:support-annotations' in project ':app'. Resolved versions for app (25.3.1) and test app (25.2.0) differ. See http://g.co/androidstudio/app-test-app-conflict for details.

Interesting that the conflict doesn't show up otherwise, but anyway...

https://stackoverflow.com/questions/28999124/resolved-versions-for-app-22-0-0-and-test-app-21-0-3-differ seems to propose a few different solutions, does anyone know which solution we should prefer?

**Comment by misaochan** [comment]
Android Studio also doesn't appear to be happy with the support library versions:

![sync_error](https://cloud.githubusercontent.com/assets/3611199/26688220/68e833c0-4735-11e7-85c9-015a9335721b.png)

**Comment by maskaravivek** [comment]
@misaochan As the accepted answer for this question suggests, excluding the conflicting module from one of the added dependencies is the right way of solving the issue. If you build the dependency tree for our app, 
```
./gradlew :app:dependencies
```

You would see this: 

<img width="793" alt="screen shot 2017-06-02 at 12 38 11 am" src="https://cloud.githubusercontent.com/assets/3069373/26696357/e3dc2d0c-472b-11e7-8e4c-163b8d648d0c.png">

Can you provide more details to reproduce the issue. :)

**Comment by misaochan** [comment]
Steps to reproduce issue:
- Create a new class subclassing the Fragment class
- Android Studio will say that Gradle project sync failed

![fragment_error](https://cloud.githubusercontent.com/assets/3611199/26715984/08e48c36-47bb-11e7-8978-1eb376e1dd53.png)

Sorry, I'm still a little confused. Adding `exclude module: 'support-annotations'` to `androidTestCompile ('com.android.support.test.espresso:espresso-core:2.2.2')` still gives me the build error:

> Error:Conflict with dependency 'com.android.support:support-annotations' in project ':app'. Resolved versions for app (25.3.1) and test app (25.2.0) differ. See http://g.co/androidstudio/app-test-app-conflict for details.

**Comment by misaochan** [comment]
One of the non-accepted answers in that thread actually works and solves the problem:

```
    configurations.all {
        resolutionStrategy.force 'com.android.support:support-annotations:25.2.0'
    }
```

But is this something we want to do?

**Comment by misaochan** [comment]
Note, I have implemented the latter fix in #707 for the time being (just so it builds and we can proceed), but after we have reached a conclusion here we should probably modify that.

**Comment by maskaravivek** [comment]
@misaochan The fix in #707 looks good to me. We can probably close the issue if others agree. :)

**Comment by misaochan** [comment]
Okay, thanks!
